@Version does not work on Temporal field without timezone information

### Expected Behavior

According to the [documentation](https://micronaut-projects.github.io/micronaut-data/latest/guide/#optimisticLocking) the @Version annotation works out of the box with any class implementing the `java.time.Temporal` interface.  Hope I'm not misunderstanding the use case for this annotation.

### Actual Behaviour

When used on a date field which holds no timezone information, the saving of the entity will fail, because Micronaut tries to assign an `OffsetDateTime` during the pre persist steps. Full stack trace:

```
io.micronaut.data.exceptions.DataAccessException: SQL Error executing INSERT: An error occurred invoking pre-persist event listeners: Specified value [2021-08-25T13:53:03.179437+02:00] is not of the correct type: class java.time.Instant
	at io.micronaut.data.runtime.operations.internal.AbstractSqlRepositoryOperations.persistOne(AbstractSqlRepositoryOperations.java:573)
	at io.micronaut.data.jdbc.operations.DefaultJdbcRepositoryOperations.lambda$persist$16(DefaultJdbcRepositoryOperations.java:709)
	at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.executeWrite(AbstractSynchronousTransactionManager.java:175)
	at io.micronaut.data.jdbc.operations.DefaultJdbcRepositoryOperations.persist(DefaultJdbcRepositoryOperations.java:707)
	at io.micronaut.data.runtime.intercept.DefaultSaveEntityInterceptor.intercept(DefaultSaveEntityInterceptor.java:46)
	at io.micronaut.data.intercept.DataIntroductionAdvice.intercept(DataIntroductionAdvice.java:80)
	at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:96)
	at io.micronaut.validation.ValidatingInterceptor.intercept(ValidatingInterceptor.java:137)
	at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:96)
	at com.example.DummyRepository$Intercepted.save(Unknown Source)
	at com.example.ScheduledJob.repeatEveryMinute(ScheduledJob.java:23)
	at com.example.$ScheduledJobDefinition$$exec1.invokeInternal(Unknown Source)
	at io.micronaut.context.AbstractExecutableMethod.invoke(AbstractExecutableMethod.java:151)
	at io.micronaut.inject.DelegatingExecutableMethod.invoke(DelegatingExecutableMethod.java:76)
	at io.micronaut.scheduling.processor.ScheduledMethodProcessor.lambda$process$5(ScheduledMethodProcessor.java:125)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: io.micronaut.data.event.PersistenceEventException: An error occurred invoking pre-persist event listeners: Specified value [2021-08-25T13:53:03.179437+02:00] is not of the correct type: class java.time.Instant
	at io.micronaut.data.runtime.event.EntityEventRegistry.prePersist(EntityEventRegistry.java:92)
	at io.micronaut.data.jdbc.operations.DefaultJdbcRepositoryOperations$JdbcEntityOperations.triggerPre(DefaultJdbcRepositoryOperations.java:1009)
	at io.micronaut.data.runtime.operations.internal.AbstractSqlRepositoryOperations$BaseOperations.triggerPrePersist(AbstractSqlRepositoryOperations.java:1538)
	at io.micronaut.data.runtime.operations.internal.AbstractSqlRepositoryOperations.persistOne(AbstractSqlRepositoryOperations.java:557)
	... 20 common frames omitted
Caused by: java.lang.IllegalArgumentException: Specified value [2021-08-25T13:53:03.179437+02:00] is not of the correct type: class java.time.Instant
	at io.micronaut.core.beans.AbstractBeanProperty.set(AbstractBeanProperty.java:144)
	at io.micronaut.data.runtime.event.listeners.VersionGeneratingEntityEventListener.prePersist(VersionGeneratingEntityEventListener.java:66)
	at io.micronaut.data.runtime.event.EntityEventRegistry$CompositeEventListener.prePersist(EntityEventRegistry.java:324)
	at io.micronaut.data.runtime.event.EntityEventRegistry.prePersist(EntityEventRegistry.java:89)
	... 23 common frames omitted
```

### Steps To Reproduce

1.  Clone my repo
2. `docker-compose up` in root folder, to start a database instance
3. Run the micronaut app
4. Wait 5 seconds until the shechuled job is executed, trying to save an entity

### Environment Information

macOS 11.3.1
openJDK 11.0.12.
Micronaut 2.5.12, but I think the same behaviour is present in Micronaut 3.0.0 as well.

### Example Application

https://github.com/uzoltan/micronaut-issue-sample

### Version

2.5.12

--- a/data-runtime/src/main/java/io/micronaut/data/runtime/event/listeners/VersionGeneratingEntityEventListener.java
+++ b/data-runtime/src/main/java/io/micronaut/data/runtime/event/listeners/VersionGeneratingEntityEventListener.java
@@ -18,19 +18,20 @@ package io.micronaut.data.runtime.event.listeners;
 import io.micronaut.core.annotation.NonNull;
 import io.micronaut.core.annotation.Nullable;
 import io.micronaut.core.beans.BeanProperty;
+import io.micronaut.core.convert.ConversionService;
 import io.micronaut.data.annotation.Version;
 import io.micronaut.data.annotation.event.PrePersist;
 import io.micronaut.data.annotation.event.PreRemove;
 import io.micronaut.data.annotation.event.PreUpdate;
-import io.micronaut.data.model.runtime.PropertyAutoPopulator;
 import io.micronaut.data.event.EntityEventContext;
 import io.micronaut.data.event.EntityEventListener;
 import io.micronaut.data.exceptions.DataAccessException;
+import io.micronaut.data.model.runtime.PropertyAutoPopulator;
 import io.micronaut.data.model.runtime.RuntimePersistentEntity;
 import io.micronaut.data.model.runtime.RuntimePersistentProperty;
 import io.micronaut.data.runtime.date.DateTimeProvider;
-
 import jakarta.inject.Singleton;
+
 import java.lang.annotation.Annotation;
 import java.time.temporal.Temporal;
 import java.util.Arrays;
@@ -92,7 +93,7 @@ public class VersionGeneratingEntityEventListener implements EntityEventListener
             throw new IllegalStateException("@Version value cannot be null");
         }
         if (Temporal.class.isAssignableFrom(type)) {
-            return dateTimeProvider.getNow();
+            return newTemporal(type);
         } else if (type == Integer.class) {
             return (Integer) previousValue + 1;
         } else if (type == Long.class) {
@@ -106,7 +107,7 @@ public class VersionGeneratingEntityEventListener implements EntityEventListener
 
     private Object init(Class<?> valueType) {
         if (Temporal.class.isAssignableFrom(valueType)) {
-            return dateTimeProvider.getNow();
+            return newTemporal(valueType);
         } else if (valueType == Integer.class) {
             return 0;
         } else if (valueType == Long.class) {
@@ -116,6 +117,11 @@ public class VersionGeneratingEntityEventListener implements EntityEventListener
         } else {
             throw new DataAccessException("Unsupported @Version type: " + valueType);
         }
+    }
+
+    private Object newTemporal(Class<?> type) {
+        Object now = dateTimeProvider.getNow();
+        return ConversionService.SHARED.convertRequired(now, type);
     }
 
 }

--- a/data-runtime/src/test/groovy/io/micronaut/data/runtime/event/listeners/VersionGeneratingEntityEventListenerSpec.groovy
+++ b/data-runtime/src/test/groovy/io/micronaut/data/runtime/event/listeners/VersionGeneratingEntityEventListenerSpec.groovy
@@ -1,0 +1,43 @@
+package io.micronaut.data.runtime.event.listeners
+
+import io.micronaut.data.annotation.AutoPopulated
+import io.micronaut.data.annotation.Id
+import io.micronaut.data.annotation.MappedEntity
+import io.micronaut.data.annotation.Version
+import io.micronaut.data.model.runtime.RuntimePersistentEntity
+import io.micronaut.data.runtime.event.DefaultEntityEventContext
+import io.micronaut.test.extensions.spock.annotation.MicronautTest
+import jakarta.inject.Inject
+import spock.lang.Specification
+
+import java.time.Instant
+
+@MicronautTest
+class VersionGeneratingEntityEventListenerSpec extends Specification {
+
+    @Inject
+    VersionGeneratingEntityEventListener entityEventListener
+
+    void "test instant version set"() {
+        when:
+            def entity = new TheEntity()
+            def mockEvent = new DefaultEntityEventContext(RuntimePersistentEntity.of(TheEntity), entity)
+            entityEventListener.prePersist(mockEvent)
+        then:
+            entity.ver
+        when:
+            entityEventListener.preUpdate(mockEvent)
+        then:
+            entity.ver
+    }
+}
+
+@MappedEntity
+class TheEntity {
+    @Id
+    @AutoPopulated
+    UUID uuid
+
+    @Version
+    Instant ver
+}

